### PR TITLE
Feat/plan trip

### PR DESCRIPTION
**Pull request by deadbelly**
This branch adds the ability to book a new trip using a form. It adds a fetch request function, some dom updating function and a whole bunch of logic to `index.js`. I haven't polished the CSS or HTML yet, or added error handling. Those are my next two steps AFTER I take a look at accessibility. I'm open to feedback about HTML and CSS refactors though.

I'd start testing this by pulling up the site (make sure the API is running too) and messing around with the Plan a Trip form. Take a look at the functions at the bottom of `index.js`, `fetchRequests.js`, and `domUpdates.js` since thats where the new logic for this branch lives.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -38,11 +42,15 @@ function fetchAndLoadDataModel() {
     .then(responses => {
       if (checkLoginCredentials(responses[0], username, password, id)) {
         generateClasses(responses[0], responses[1], responses[2]);
+        displayAllTrips();
+        domUpdates.displaySidebar(user, sidebar);
```
Comment (JeffKersting) [inline src/index.js:46]: Possibly try to break this code into smaller functions that perform a specific login display page function?

File: src/index.js [hunk 2]
```diff
@@ -38,11 +42,15 @@ function fetchAndLoadDataModel() {
...
         domUpdates.toggleHidden(loginPage);
         domUpdates.toggleHidden(main);
-        domUpdates.toggleHidden(navBar);
-        displayAllTrips();
-        domUpdates.displayNavBar(user, navBar);
+        domUpdates.toggleHidden(sidebar);
+        setStartMin()
+        document.querySelector('.plan-trip-button').addEventListener('click', toggleFormView)
```
Comment (JeffKersting) [inline src/index.js:51]: Could add event listener to page that fires functions passing the event target in to DRY up some of the code.

File: src/domUpdates.js [hunk 3]
```diff
@@ -24,15 +24,52 @@ const domUpdates = {
...
+  },
+
+  displayNewTripForm(destinations) {
+    document.querySelector('.new-trip-form').insertAdjacentHTML('afterbegin',
+    `<select class="destination-list" name="">
+    </select>
+    <input type="date" class="trip-start" min="${new Date().toISOString().substring(0, 10)}">
+    <input type="date" class="trip-end" min="${new Date().toISOString().substring(0, 10)}">
+    <input type="number" class="num-travelers" value="1" min="1">
+    <h3></h3>
```
Comment (JeffKersting) [inline src/domUpdates.js:51]: Is this <h3> being used for styling? I could have missed where something is being inserted.

Comment (JeffKersting) [reply src/domUpdates.js:51]: h3 tag...

File: src/domUpdates.js [hunk 4]
```diff
@@ -24,15 +24,52 @@ const domUpdates = {
     </article>`);
   },
 
-  displayNavBar(user, navBar){
-    navBar.insertAdjacentHTML('afterbegin',
+  clearTrips(tripList) {
```
Comment (JeffKersting) [inline src/domUpdates.js:27]: Can this be incorporated into the click event handler?

File: src/domUpdates.js (hunk 3 above)
Comment (deadbelly) [reply src/domUpdates.js:51]: Oh no this function needs to get deleted! I moved it to `index.html` and fixed the `<h3>` over there.

**Review by JeffKersting — APPROVED** [review]
Code is looking really good! Really like the way dates are handled. I noticed there is a lot of toggle hidden happening within code blocks, could try to create a single function to handle the major ones (like hiding the login page/displaying the user page after login), just to shorten up the code block. But aside from a few minor refactoring things, the logic looks really good! Easy to follow the process.

**Pull request merged**